java/src/jmri/jmrix/mqtt/MqttPowerManager.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * MqttPowerManager.java
+ *
+ * PowerManager implementation for controlling layout power
+ *
+ * @author Dean Cording Copyright (C) 2023
+ *
+ */
+package jmri.jmrix.mqtt;
+
+import jmri.JmriException;
+import jmri.managers.AbstractPowerManager;
+
+import javax.annotation.Nonnull;
+import jmri.JmriException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MqttPowerManager extends AbstractPowerManager<MqttSystemConnectionMemo> implements MqttEventListener {
+
+    private static final String onText = "ON";
+    private static final String offText = "OFF";
+    private final MqttAdapter mqttAdapter;
+
+    public MqttPowerManager(MqttSystemConnectionMemo memo) {
+        super(memo);
+        mqttAdapter = memo.getMqttAdapter();
+
+    }
+
+    public void setSendTopic(@Nonnull String sendTopic) {
+        this.sendTopic = sendTopic;
+    }
+    public void setRcvTopic(@Nonnull String rcvTopic) {
+        this.mqttAdapter.unsubscribe(rcvTopic, this);
+        this.rcvTopic = rcvTopic;
+        this.mqttAdapter.subscribe(rcvTopic, this);
+
+        log.info("Subscribed to {}", rcvTopic);
+
+    }
+
+    @Nonnull
+    public String sendTopic = "track/power"; // for constructing topic; public for script access
+    @Nonnull
+    public String rcvTopic = "track/power"; // for constructing topic; public for script access
+
+    @Override
+    public boolean implementsIdle() {
+        return false;
+    }
+
+    @Override
+    public void setPower(int v) throws JmriException {
+        if (v == ON) {
+            // send TRACK_POWER_ON
+            sendMessage(onText);
+        } else if (v == OFF) {
+            // send TRACK_POWER_OFF
+            sendMessage(offText);
+        }
+    }
+
+
+
+    // to free resources when no longer used
+    @Override
+    public void dispose() throws JmriException {
+    }
+
+
+     private void sendMessage(String c) {
+        jmri.util.ThreadingUtil.runOnLayoutEventually(() -> {
+            mqttAdapter.publish(this.sendTopic, c);
+        });
+        log.debug("sent {}", c);
+    }
+
+    @Override
+    public void notifyMqttMessage(String receivedTopic, String message) {
+        if (!receivedTopic.endsWith(rcvTopic)) {
+            log.error("Got a message whose topic ({}) wasn't for me ({})", receivedTopic, rcvTopic);
+            return;
+        }
+        log.debug("notifyMqttMessage with {}", message);
+
+        // handle on/off
+        try {
+            switch (message) {
+                case onText: super.setPower(ON); break;
+                case offText: super.setPower(OFF); break;
+            }
```

Comment:
Static analysis failes on this line due to that this switch case statement should have a default case. If the default case should never happen, do a `log.error()`.

Reply:
Fixed